cdk-project/lib/images/codebuild-image/python/src/notebooks/parse.py

```diff
@@ -225,3 +258,60 @@ def skip(notebook):
     elif any([str(directory) in SKIP_LIST for directory in directories]):
         return True
     return False
+
+
+def local_mode_mandatory(notebook):
+    """Check whether the notebook should be skipped because runs only in Local Mode
+
+    Args:
+        notebook (Path): The notebook to check.
+
+    Returns:
+        bool: True if local mode is mandatory.
+
+    """
+    directories = Path(notebook).parents
+    return "reinforcement_learning" in directories and str(notebook) not in get_lm_optional_nb_names()
+
+
+def uses_unsupported_feature_or_framework(notebook,skip_args):
+    """Check whether the notebook should be skipped because it uses FXS, Docker or Local Mode.
+
+    Args:
+        notebook (Path): The notebook to check.
+        skip_args (dict): A dictionary containing feature flag for each kind of skip, user-driven with sensible defaults
+    Returns:
+        bool: True if the notebook uses FXS, Docker or Local Mode.
+
+    """
+    functionalities_to_check = {
+        "docker": ["docker\s+", "docker-compose\s+"],
+        "local_mode": ['instance_type\s?=\s?"local"'],
+        "fsx_efs": ['\s?FSxLustre\s?', '\s?EFS\s?']
```

Comment:
Please replace second "?" with "+".
do we  always expect space before above words as they can be substring of string